### PR TITLE
Put warrant issuance behind feature unstable-warrants

### PR DESCRIPTION
**Pull request by maackle**
### Summary
Proposal to put warranting behind a feature. Everything "write"-related is put behind the feature, i.e. sys validation does not issue warrants, app validation does not issue warrants, and the functions in holochain_state to insert warrants into the DB are behind the feature as well, just in case. It's up for discussion if forks should be detected even without the feature.

### TODO:
- [x] update CI build/test to include proper feature set
- [x] CHANGELOGs updated with appropriate info
- [x] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

**Comment by maackle** [comment]
@zippy @holochain/core-dev is this approach desired? I'm not expecting to finish this work because the general sense was that it's too invasive. Should I close this or leave it open?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ concurrency:
 
 env:
   CARGO_INCREMENTAL: 0
-  RUSTFLAGS: '-Dwarnings'
+  RUSTFLAGS: "-Dwarnings"
```
Comment (jost-s) [inline .github/workflows/test.yml:15]: This is my formatter - should I revert? Add it to CI?

File: .github/workflows/test.yml
Comment (jost-s) [inline .github/workflows/test.yml:1]: Copied from @ThetaSinner's sharding PR.

File: Makefile
Comment (jost-s) [inline Makefile:1]: Ditto, copied

File: crates/holochain/src/core/workflow/sys_validation_workflow.rs [hunk 2]
```diff
@@ -231,9 +231,9 @@ pub async fn sys_validation_workflow<Network: HolochainP2pDnaT + 'static>(
 async fn sys_validation_workflow_inner(
     workspace: Arc<SysValidationWorkspace>,
     current_validation_dependencies: SysValDeps,
-    network: &impl HolochainP2pDnaT,
-    keystore: MetaLairClient,
-    representative_agent: AgentPubKey,
+    _network: &impl HolochainP2pDnaT,
+    _keystore: MetaLairClient,
+    _representative_agent: AgentPubKey,
```
Comment (jost-s) [inline crates/holochain/src/core/workflow/sys_validation_workflow.rs:236]: workaround to not use #[cfg(feature...)} here

File: crates/holochain/src/core/workflow/sys_validation_workflow.rs [hunk 3]
```diff
@@ -302,126 +302,186 @@ async fn sys_validation_workflow_inner(
...
-                    // Author a ChainFork warrant if fork is detected
-                    let action = chain_op.action();
-                    if let Some(forked_action) = detect_fork(txn, &action)? {
-                        let signature = chain_op.signature().clone();
-                        let action_hash = action.to_hash();
-                        forked_pairs.push((
-                            action.author().clone(),
-                            ((action_hash, signature), forked_action),
-                        ));
+    #[cfg(not(feature = "unstable-warrants"))]
```
Comment (jost-s) [inline crates/holochain/src/core/workflow/sys_validation_workflow.rs:305]: I thought of leaving this code of detecing forks and invalid ops and modifying the summary untouched and just put the insertion of warrants into the DB behind the feature. That would be easier to read but more mixed with warranting.

File: crates/holochain/src/core/workflow/sys_validation_workflow/tests.rs [hunk 4]
```diff
@@ -266,8 +270,11 @@ async fn run_test(
     let (bad_update_action, bad_update_entry_hash, link_add_hash) =
         bob_makes_a_large_link(&bob_cell_id, &conductors[1].raw_handle(), &dna_file).await;
 
-    // Integration should have 14 chain ops in it + 1 warrant op + the running tally
+    // Integration should have 14 chain ops in it + 1 warrant op (if unstable-warrants enabled) + the running tally
+    #[cfg(feature = "unstable-warrants")]
```
Comment (jost-s) [inline crates/holochain/src/core/workflow/sys_validation_workflow/tests.rs:274]: This tests much more than just warrants, so I thought to keep it.

File: crates/holochain_cascade/tests/tests/get_activity.rs [hunk 5]
```diff
@@ -598,6 +592,32 @@ async fn test_must_get_agent_activity(
...
+#[tokio::test(flavor = "multi_thread")]
+async fn test_must_get_agent_activity_with_warrants(
+    data: Data,
+    author: AgentPubKey,
+    filter: ChainFilter,
+) -> MustGetAgentActivityResponse {
+    test_must_get_agent_activity_inner(data, author, filter).await
+}
+
+async fn test_must_get_agent_activity_inner(
```
Comment (jost-s) [inline crates/holochain_cascade/tests/tests/get_activity.rs:617]: Splitting out a shared function to not have to duplicate all code and also run the non-warrant tests regardless of the feature.

File: .github/workflows/test.yml (hunk 1 above)
Comment (ThetaSinner) [reply .github/workflows/test.yml:15]: Add it to CI :) Not necessarily on this PR but we already did that on Wind Tunnel

File: crates/holochain/CHANGELOG.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
 - **BREAKING**: As the DPKI feature is unstable and incomplete, it is disabled with default cargo features and put behind a feature called `unstable-dpki`. If this feature is specified at compile time, DPKI is enabled by default.
-- **BREAKING**: Conductor::get_dna_definitions now returns an `IndexMap` to ensure consistent ordering.
+- **BREAKING**: Issuing and persisting warrants is behind a feature `unstable-warrants` now. Warrants have not been tested extensively and there is no way to recover from a warrant. Hence the feature is considered unstable and must be explicitly enabled.
```
Comment (ThetaSinner) [inline crates/holochain/CHANGELOG.md:10]: I'd also mention that warrants get stored in the Actions table and some queries get broken when warrants are stored

**Review by ThetaSinner — APPROVED** [review]
Looks good!

File: crates/holochain/CHANGELOG.md (hunk 6 above)
Comment (jost-s) [reply crates/holochain/CHANGELOG.md:10]: So that doesn't work right now?

Comment (jost-s) [reply crates/holochain/CHANGELOG.md:10]: I mean when a warrant action is stored, does a part of Holochain break is how I understand this?

Comment (ThetaSinner) [reply crates/holochain/CHANGELOG.md:10]: Yes, the `query` function which retrieves your source chain will just give you a deserialisation error

Comment (ThetaSinner) [reply crates/holochain/CHANGELOG.md:10]: I'm assuming that's probably not all,  it's just all I know about from Holo testing so far

Comment (jost-s) [reply crates/holochain/CHANGELOG.md:10]: Oh, that's pretty bad and another reason to put this behind an unstable feature.

**Pull request merged**